core-bundle/src/Resources/contao/library/Contao/Controller.php

```diff
@@ -1132,6 +1133,23 @@ public static function redirect($strLocation, $intStatus=303)
 		throw new RedirectResponseException($strLocation, $intStatus);
 	}
 
+	/**
+	 * Get the current request from the request stack
+	 *
+	 * @return Request
+	 */
+	public static function getCurrentRequest()
```

Comment:
Not sure if you like the new method here. I wanted to save myself from copying the exception multiple times.